lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedSystemAccessControl.java

```diff
@@ -994,6 +1007,11 @@ private boolean checkTablePermission(
         return false;
     }
 
+    private static SystemSecurityContext getSecurityContextForPrincipal(SystemSecurityContext context, TrinoPrincipal principal)
+    {
+        return new SystemSecurityContext(Identity.forUser("").withEnabledRoles(Set.of(principal.getName())).build(), context.getQueryId());
```

Comment:
Can't you get the user from `SystemSecurityContext.Identity`?

Reply:
The goal is to see if the target role has the privilege; if we use the original user here, we might check the wrong thing, i.e. if the *current user* does have the privilege, which is exactly what we want to avoid.